_gsocblogs/2022/blog_Rucio_NimishBongale.md

```diff
@@ -1,127 +1,279 @@
 
-## Coding Period
+## Coding Period Deliverables
 
-- Understand Rucio WebUI in greater depth and their existing components.
-- Get a sample storybook running and deployed.
-- Interact with mentors and other team members, and establish a rapport.
-- Learn StorybookJS & NextJS
+- Make new custom built Rucio-flavoured components.
+- Integrate React, Typescript, Webpack & Storybook.
+- Integrate the Rucio UI with our custom built components itself, have it up and running.
+- Migrate old JS code to new TS code.
+- Have Rucio component + usage guides on storybook.
+- Finalize on content across various screens.
+- Complete development on high priority UI screens and have them migrated.
+- Carry out the necessary integration.
+- Survey development builds for a11y using WAVE and Accessibility Insights for Web.
+- Survey for responsiveness using Chrome Dev Tools.
+- Generate Lighthouse reports.
+- Write a blog post summarizing everything, and about my performance cum involvement in Rucio.
+- Seek daily approvals until everything is finalised.
+- Go code upteen times for any minuscule errors.
+- Clean code, make sure everything is properly linted and ready before the final closing commits.
+- Publish my endterm blog post, rounding up my wholesome journey.
+- Seek final approval.
+- Present the users with a new Dashboard View.
+- Promote the use of CERN attributed works.
+- Interact with the community, answer queries or doubts regarding Rucio.
+- Carry out maintenance work of the repositories I’ve contributed to.
+- Keep resolving relevant issues in the Rucio repositories, making contributions if necessary.
+- Leverage experience gained during this period for future endeavors.
 
----
+## Project Milestones
 
-- Tackle queries regarding the page structure etc. and seek approval for the same.
-- Begin to code the content which will need to fill up the Rucio landing site.
-- Start replacing templating with the RESTful approach. Mock data and carry out a POC for authorization workflows.
+### Pages
 
----
+In total 8 pages were created as a part of this endeavour.
 
-- Finalize the approach.
-- Keep the code ready for accepting storybook UI contents.
-- Completely replace templating with modern TSX code.
+#### Login Page
 
----
+The Login page serves as the entry point to the entire application. It makes use of the custom built Form component, which supports field validation. It handles multiple auth mechanisms such as Username/Password, x509 ceritficate, OIDC & OIDC PKCE. A successful sign in leads us to the Home page, while the user is notified in case of an unsuccessful one.
```

Comment:
You could also mention some of the challenges with OIDC, PKCE and x509 strategies somewhere in this document.